azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -239,6 +321,51 @@ def patch_twin_reported_properties(self, reported_properties_patch):
     def receive_twin_desired_properties_patch(self):
         pass
 
+    def update_sastoken(self, sastoken):
+        """
+        Update the client's SAS Token used for authentication.
+
+        This API can only be used if the client was initially created with a SAS Token.
+
+        :param str sastoken: The new SAS Token string for the client to use
+
+        :raises: ValueError if the sastoken parameter is invalid
+        :raises: :class:`azure.iot.device.exceptions.ClientError` if the client was not initially
+            created with a SAS token.
+        """
+        if not isinstance(
+            self._mqtt_pipeline.pipeline_configuration.sastoken, st.NonRenewableSasToken
```

Comment:
>pipeline_configuration.sastoken [](start = 32, length = 31)

what is the pipeline was configured with x509 auth?  is pipeline_configuration.sastoken None, or could this raise an AttributeError?

Reply:
It's None

---
In reply to: [506688919](https://github.com/Azure/azure-iot-sdk-python/pull/692#discussion_r506688919) [](ancestors = 506688919)